### ISSUE TITLE
TSInstall templ doesn't work until adding plugin (the old way), and even after TS installs templ successfully, it doesn't seem to work

### ISSUE THREAD

**Issue by nathan-hello** [vrischmann/tree-sitter-templ#14]
Running `:TSInstall templ` gives me the following:
```
Installation not possible: ...vim/lazy/nvim-treesitter/lua/nvim-treesitter/install.lua:86: Parser not available for language "templ"
See https://github.com/nvim-treesitter/nvim-treesitter/#adding-parsers on how to add a new parser!
```

I went through the old readme, before it was officially available on tree-sitter, and it said to install it as a package and set up treesitter that way. This is the config I made, as per the old docs. I'm using LazyVim.

```lua
require("lazy").setup({
-- other plugins, including tree-sitter
{ "vrischmann/tree-sitter-templ", config = function() require("tree-sitter-templ").setup({}) end }                  -- https://github.com/vrischmann/tree-sitter-templ/commit/2a8326a0ee921c5ed24b3e627283390b7bf1d64f
```

Then, later:

```lua
require("nvim-treesitter.configs").setup({ auto_install = true, highlight = { enabled = true, additional_vim_regex_highlighting = true }, })

require("luasnip")

local treesitter_parser_config = require "nvim-treesitter.parsers".get_parser_configs()
treesitter_parser_config.templ = {
  install_info = {
    url = "https://github.com/vrischmann/tree-sitter-templ.git",
    files = {"src/parser.c", "src/scanner.c"},
    branch = "master",
  },
}

vim.treesitter.language.register('templ', 'templ')
```

Even then, I don't get highlighting and everything else. 

`:checkhealth` gives me this

```
Parser/Features         H L F I J
...other modules
  - templ               ✓ . . . ✓
```

but `:TSModuleInfo` gives me this:
```
>> default          context_commentstring  highlight  illuminate  incremental_selection  indent  
...other modules
templ               ✗                      ✗          ✗           ✗                      ✗       
```

I'm at a bit of a loss. Is there something I'm missing? I'm not sure if this program incorporates templ's builtin lsp, but even so I just wanted to get syntax highlighting to work

**Comment by knudtty** [comment]
I have a similar issue - I had to update treesitter to the most recent version, and eventually `:TSInstall templ` found the package and installed. My `:checkhealth` shows the same as yours, but I get no syntax highlighting. I also can't get the lsp to work at all.

**Comment by knudtty** [comment]
Whoops, just found my issue. I had to add the [templ.vim](https://github.com/joerdav/templ.vim) package. Then, the templ filetype still wasn't recognized, so I had to add
```lua
vim.filetype.add({
  pattern = {
    ["*.templ"] = "templ",
  },
})
```
somewhere in my config. Everything works as expected now.

**Comment by vrischmann** [comment]
Hi.

@nathan-hello I would guess that something is wrong with your nvim-treesitter installation because templ is definitely available [by default](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lua/nvim-treesitter/parsers.lua#L2160-L2166) since a couple of weeks ago. Did you update nvim-treesitter ?

@knudtty You shouldn't need templ.vim, I don't have it installed. But you do need to add the filetype yes.

**Comment by nathan-hello** [comment]
@vrischmann Thanks for the quick reply. I updated all of my plugins, and templ was installing properly. It still wasn't working, but I noticed `:InspectTree` was giving me a good tree-sitter looking tree, so it's obviously parsing. Running `:TSBufEnable highlight` was the solution, as per this part of the [tree-sitter readme](https://github.com/nvim-treesitter/nvim-treesitter#i-experience-weird-highlighting-issues-similar-to-78). So, this is what I put in my init.lua

```lua
vim.filetype.add({ extension = { templ = "templ" } })
vim.api.nvim_create_autocmd("BufEnter", {
    pattern = "*.templ",
    callback = function() vim.cmd("TSBufEnable highlight") end
})
```

Syntax highlighting is working, without the older, vimscript plugin (though I did try it and it did work) and without the tree-sitter plugin that you made before it got added officially.  

I still have a few questions about the templ lsp, but I know that's not your side of the house. Thanks for making the parser and for helping me.
